assertj-parent/pom.xml

```diff
@@ -205,6 +205,16 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-jar-plugin</artifactId>
           <version>${maven-jar-plugin.version}</version>
+          <configuration>
+            <archive>
+              <index>true</index>
```

Comment:
Hi @ascopes, I'm not familiar with the archive index so I tried to educate myself.

The following description from the [Oracle documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/jar/jar.html#JAR_Index) captured my attention:

> Once the class loader finds a INDEX.LIST file in a particular jar file, it always trusts the information listed in it. If a mapping is found for a particular class, but the class loader fails to find it by following the link, an InvalidJarIndexException is thrown. When this occurs, the application developer should rerun the jar tool on the extension to get the right information into the index file.

Also, [SO-16430146](https://stackoverflow.com/questions/16430146/) and [SO-2020800](https://stackoverflow.com/questions/2020800/) contained some real-life usage of index files.

Overall, I feel the archive index is meant to be provided by the final application and not by libraries.

Do I understand it wrongly?

I was inspecting the JARs from projects like JUnit, Mockito, and Spring, and I couldn't find any index file in there.

Reply:
This is a fair point actually, will update to omit this

Reply:
updated!